XcodeServerSDKTests/XcodeServerEndpointsTests.swift

```diff
@@ -24,5 +32,291 @@ class XcodeServerEndpointsTests: XCTestCase {
+    }
+    
+    func testEndpointURLCreationForBotsBotRev() {
+        let paramsArray = [
+            [
+                "rev": "revValue",
+                "bot": "botValue"
+            ],
+            [
+                "otherKey": "otherValue",
+                "bot": "botValue",
+                "rev": "revValue"
+            ]
+        ]
+        for params in paramsArray {
+            let url = self.endpoints?.endpointURL(.Bots, params: params)
+            XCTAssertEqual(url!, "/api/bots/botValue/revValue", "endpointURL(.Bots, \(params)) should return \"/api/bots/botValue/revValue\"")
+        }
+    }
+    
+    // MARK: endpointURL(.Integrations)
+    
+    func testEndpointURLCreationForIntegrationsWithoutParams() {
+        let url = self.endpoints?.endpointURL(.Integrations)
+        XCTAssertEqual(url!, "/api/integrations", "endpointURL(.Integrations) should return \"/api/integrations\"")
+    }
+    
+    func testEndpointURLCreationForBotsBotIntegrations() {
+        let paramsArray = [
+            [
+                "bot": "botValue"
+            ],
+            [
+                "otherKey": "otherValue",
+                "bot": "botValue"
+            ],
+            [
+                "otherKey": "otherValue",
+                "bot": "botValue",
+                "integration": "integrationValue"
+            ]
+        ]
+        for params in paramsArray {
+            let url = self.endpoints?.endpointURL(.Integrations, params: params)
+            XCTAssertEqual(url!, "/api/bots/botValue/integrations", "endpointURL(.Integrations, \(params)) should return \"/api/bots/botValue/integrations\"")
+        }
+    }
+    
+    func testEndpointURLCreationForBotsBotRevIntegrations() {
+        let paramsArray = [
+            [
+                "rev": "revValue",
+                "bot": "botValue"
+            ],
+            [
+                "otherKey": "otherValue",
+                "bot": "botValue",
+                "rev": "revValue"
+            ],
+            [
+                "otherKey": "otherValue",
+                "bot": "botValue",
+                "rev": "revValue",
+                "integration": "integrationValue"
+            ]
+        ]
+        for params in paramsArray {
+            let url = self.endpoints?.endpointURL(.Integrations, params: params)
+            XCTAssertEqual(url!, "/api/bots/botValue/revValue/integrations", "endpointURL(.Integrations, \(params)) should return \"/api/bots/botValue/revValue/integrations\"")
+        }
+    }
+    
+    func testEndpointURLCreationForIntegrationsIntegration() {
+        let paramsArray = [
+            [
+                "integration": "integrationValue",
+            ],
+            [
+                "otherKey": "otherValue",
+                "integration": "integrationValue"
+            ],
+            [
+                "rev": "revValue",
+                "otherKey": "otherValue",
+                "integration": "integrationValue"
+            ]
```

Comment:
```
[
    "integration": "integrationValue",
]
```
This is the desired set of parameters when it comes to `/api/integrations/integrationValue` path creation.

```
[
    "otherKey": "otherValue",
    "integration": "integrationValue"
]
```
In this case test checks if additional/random parameter won't affect path creation.

```
[
    "rev": "revValue",
    "otherKey": "otherValue",
    "integration": "integrationValue"
]
```
This case was meant to test if adding `rev` key without `bot` key won't add `revValue` to the path, but it can be removed since `testEndpointURLCreationForBots` tests it.